src/System.IO.Pipes/src/System/IO/Pipes/PipeStream.cs

```diff
@@ -374,6 +374,13 @@ public override void Flush()
             }
         }
 
+        public override Task FlushAsync(CancellationToken cancellationToken)
+        {
+            // Just throw exceptions -- we do no I/O in this method.
+            Flush();
+            return Task.CompletedTask;
+        }
```

Comment:
This changes the behavior of PipeStream.FlushAsync if Flush throws an exception.  Previously such exceptions would have been stored into the returned Task; now they'll propagate out synchronously.

Reply:
I haven't been able to notice a consistent behavior in corefx with regard to letting exceptions be thrown from Task-returning methods. It seems they're allowed, in general. But for backward compat concerns I can see the risk. Shall I prepare a fix?

Reply:
>I haven't been able to notice a consistent behavior in corefx with regard to letting exceptions be thrown from Task-returning methods.

In general the rule is that only usage exceptions should be allowed to propagate out synchronously, e.g. ArgumentExceptions are thrown synchronously, InvalidOperationExceptions are returned in the Task.  Most coreclr/corefx usage follows that.  Of course, with lots of different people working on the codebases, cases slip through, and once we've shipped them, it's generally not worth it to change them.

> Shall I prepare a fix?

That'd be good, thanks.  Basically just:
```C#
try
{
    Flush();
    return Task.CompleteTask;
}
catch (Exception exc)
{
    return Task.FromException(exc);
}
```

Thanks.

Reply:
Done https://github.com/dotnet/corefx/pull/32432